### PR TITLE
[iceoryx] Fix acl dependency linking. Fix feature description

### PR DESCRIPTION
**Pull request by ex-purple**
1. fix wrong acl linking (see problem in  https://github.com/microsoft/vcpkg/pull/29150)
2. fix wrong description for the "many-to-many" feature

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
I want to double check this with folks with a better understanding of cmake's behavior here; will keep you posted.

File: ports/iceoryx/acl.patch [hunk 1]
```diff
@@ -1,24 +1,36 @@
...
+ )
+
+ if(LINUX)
+-    target_link_libraries(iceoryx_hoofs PRIVATE acl atomic ${CODE_COVERAGE_LIBS})
++    target_link_directories(iceoryx_hoofs PUBLIC $<BUILD_INTERFACE:${ACL_LIBRARY_DIRS}>)
++    target_link_libraries(iceoryx_hoofs
++    PRIVATE
++    $<BUILD_INTERFACE:${ACL_LIBRARIES}>
++    atomic
++    ${CODE_COVERAGE_LIBS})
```
Comment (BillyONeal) [inline ports/iceoryx/acl.patch:15]: I don't think this is actually a fix. For example, to my understanding, `PUBLIC + BUILD_INTERFACE` should be more or less equivalent to `PRIVATE`. Moreover, the problem you describe as trying to solve here is that a dependent port, `cyclonedds`, didn't get forwarded acl link information correctly. That means it needs to be public information on the target, not private.

What do you think of https://github.com/BillyONeal/vcpkg/commit/e90a4c2d1e37e15f60154f7daf9f8cba0960483e as a fix instead?

**Review by BillyONeal — CHANGES_REQUESTED** [review]

File: ports/iceoryx/acl.patch (hunk 1 above)
Comment (ex-purple) [reply ports/iceoryx/acl.patch:15]: I answered in this thread

Comment (ex-purple) [reply ports/iceoryx/acl.patch:15]: I agree with these changes, although I'm not very happy with the explicit ACL dependency

**Pull request closed without merging**